sdk/search/search-documents/src/searchIndexClient.ts

```diff
@@ -159,18 +160,43 @@ export class SearchIndexClient {
     }
   }
 
+  private async *listIndexesAll(options: ListIndexesOptions = {}): AsyncIterableIterator<Index> {
+    for await (const page of this.listIndexesPage(options)) {
+      yield* page;
+    }
+  }
+
   /**
-   * Retrieves a list of names of existing indexes in the service.
+   * Retrieves a list of existing indexes in the service.
    * @param options Options to the list index operation.
    */
-  public async listIndexesNames(options: ListIndexesOptions = {}): Promise<Array<string>> {
-    const { span, updatedOptions } = createSpan("SearchIndexClient-listIndexesNames", options);
+  public listIndexes(options: ListIndexesOptions = {}): IndexIterator {
+    const iter = this.listIndexesAll(options);
+
+    return {
+      next() {
+        return iter.next();
```

Comment:
I'm not sure how this currently manifests, but the `next` property/function should be null for now so `asPages` doesn't even try to call back out to the service (will likely fail, currently).

Reply:
Since the iterator does `yield*` on the array, it yields each element one at a time. Since the function terminates with no further yields after that, it will return an `Iterable` with the value as `undefined` and `done` as true.

There's no code path that re-calls fetching results right now, but we can add one later.

Reply:
For the paged version I yield the one array we have from fetching results right now and then it also finishes